src/clj/fluree/db/query/exec/where.cljc

```diff
@@ -566,6 +566,22 @@
                           clause-ch)
     out-ch))
 
+(defmethod match-pattern :values
+  [db fuel-tracker solution pattern error-ch]
+  (let [inline-solutions (pattern-data pattern)
+        ;; need to remove ::sids from the solution as those may differ based on which graph they are from
+        solution* (reduce (fn [soln [var match]]
+                            (assoc soln var (cond-> (dissoc match ::sids)
```

Comment:
I think this is still brittle and could unintentionally break if we add more information to the matches. Instead of removing known divergent information from the solution and doing a top level equality check, I think we should reduce over the variables in the inline solution and explicitly check if the value and datatype of the match bound to that variable in the solution is the same as the value and datatype of the inline solution and, if there is no value/datatype, we should check if it's a matched iri and the iris are the same. The `get-value` and `get-datatype` functions will help you in the first case, and the `get-iri` will help you in the second. 

We should explicitly check the things we know must be the same, instead of explicitly eliminating the things we think might be different, because we could add more things that might be different later.

Reply:
Good point, I've updated this section (and discovered the very handy `update-vals` core fn, thanks Rich) to select the important bits.